task-driver/src/create_new_wallet.rs

```diff
@@ -47,8 +46,11 @@ const NEW_WALLET_TASK_NAME: &str = "create-new-wallet";
 pub struct NewWalletTask {
     /// The wallet to create
     pub wallet: StateWallet,
-    /// A starknet client for the task to submit transactions
-    pub starknet_client: StarknetClient,
+    /// The proof of `VALID WALLET CREATE` for the wallet, generated in the
+    /// first step
+    pub proof: Option<ValidWalletCreateBundle>,
```

Comment:
nit: can we call this `proof_bundle` instead of `proof`, it removes confusion around how and when the statement gets submitted to the contract